plugins/modules/zos_blockinfile.py

```diff
@@ -666,19 +666,28 @@ def main():
 
     marker = "{0}\\n{1}\\n{2}".format(marker_begin, marker_end, marker)
     block = transformBlock(block, ' ', indentation)
+    gdg = False
     # analysis the file type
+    if "/" not in src:
+        dataset = data_set.MVSDataSet(
+            name=src
+        )
+        src = dataset.name
+        gdg = dataset.name
+
+    if data_set.DataSet.is_gds_relative_name(src) and gdg is False:
+        module.fail_json(msg="{0} does not exist".format(src))
```

Comment:
Is this error message correct? Seems to me that if src contains "/" and is a GDS relative name then would fail. Let's say we have for some reason something like "OMVSADM.A/A.(+1)" which seems to be the case here. Isn't this catch by the betterargparser? Also, if this is failing when a GDS relative name is found wouldn't that get resolved by line 672? Could you share a failing example for this case I just can't wrap my head around a failing scenario.

Reply:
This case with the catch of relative name suggest that is a (+1) GDG and blockinfile and lineinfile required datasets already created, so if still as relative name throw the error

Reply:
cool